code/game/atoms.dm

```diff
@@ -563,6 +563,25 @@
 					for(var/datum/reagent/R in reagents.reagent_list)
 						total_volume += R.volume
 					. += "[total_volume] units of various reagents"
+				//-------- Beer goggles ---------
+				if(user.can_see_boozepower())
+					var/total_boozepower = 0
+					var/list/taste_list = list()
+					if(length(reagents.reagent_list))
+						for(var/datum/reagent/consumable/ethanol/B in reagents.reagent_list)
+							var/real_boozepower = B.boozepwr
+							if(real_boozepower<0) // minus booze power is reversed to light drinkers, but is actually 0 to normal drinkers.
+								real_boozepower = 0
+							total_boozepower += B.volume*real_boozepower
+						for(var/datum/reagent/R in reagents.reagent_list)
+							if(istype(R, /datum/reagent/consumable/ethanol/fruit_wine) && !(user.stat == DEAD)) // taste of fruit wine is mysterious, but can be known by ghosts
```

Comment:
Why is fruit wine specifically having no taste? Make a var on reagent instead

Reply:
because it can have various tastes based on which fruit you fermented, but I didn't want them to know it earlier before they actually taste it than just checking its examine text.